inst/shiny-server/server/upload.R

```diff
@@ -200,34 +231,38 @@ observeEvent(selected_folders(), {
                    Please ensure the data is not corrupted.")
     return(NULL)
   }
-  else if (attr(DataList$data, 'suite') == NEVERGRAD) {
+  update_menu_visibility(attr(DataList$data, 'suite'))
+  set_format_func(attr(DataList$data, 'suite'))
+  set_color_scheme("Default", get_algId(DataList$data))
+})
+
+update_menu_visibility <- function(suite){
+  if (suite == NEVERGRAD) {
     #TODO: Better way of doing this such that these pages are not even populated with data instead of just being hidden
-    session$sendCustomMessage(type = "manipulateMenuItem", message = list(action = "hide", tabName = "RT_ECDF"))
-    session$sendCustomMessage(type = "manipulateMenuItem", message = list(action = "hide", tabName = "ERT_convergence"))
-    session$sendCustomMessage(type = "manipulateMenuItem", message = list(action = "hide", tabName = "ERT_data"))
-    session$sendCustomMessage(type = "manipulateMenuItem", message = list(action = "hide", tabName = "ERT"))
-    session$sendCustomMessage(type = "manipulateMenuItem", message = list(action = "hide", tabName = "RT_PMF"))
-    session$sendCustomMessage(type = "manipulateMenuItem", message = list(action = "hide", tabName = "RT_PARAMETER"))
-    session$sendCustomMessage(type = "manipulateMenuItem", message = list(action = "hide", tabName = "RT_Statistics"))
+    session$sendCustomMessage(type = "manipulateMenuItem", message = list(action = "hide", tabName = "#shiny-tab-ERT"))
+    # session$sendCustomMessage(type = "manipulateMenuItem", message = list(action = "hide", tabName = "ERT_convergence"))
+    # session$sendCustomMessage(type = "manipulateMenuItem", message = list(action = "hide", tabName = "ERT_data"))
```

Comment:
There is no need to hide the remaining tabs for Nevergrad. Or you renamed them?

Reply:
This is because of the nested structure of the tabs, just hiding the root automatically means the children are not accessible either (in combination with the earlier line of javascript this finally seems to work correctly)